### ISSUE TITLE
'NoneType' object is not callable

### ISSUE THREAD

**Issue by International** [xolox/python-coloredlogs#33]
Hi Peter, I've noticed the following exception happening:

```
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/logging/__init__.py", line 859, in emit
    msg = self.format(record)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/logging/__init__.py", line 732, in format
    return fmt.format(record)
  File "/Users/geo/code/cx/lib/python2.7/site-packages/coloredlogs/__init__.py", line 891, in format
    copy = Empty()
TypeError: 'NoneType' object is not callable
```

I'm using a 3rd party package, `happybase`, and in the `__del__` method, it's calling `close`, which is doing this:

```python
    def close(self):
        """Close the underyling transport to the HBase instance.

        This method closes the underlying Thrift transport (TCP connection).
        """
        if not self.transport.is_open():
            return

        if logger is not None:
            # If called from __del__(), module variables may no longer
            # exist.

            logger.debug(
                "Closing Thrift transport to %s:%d",
                self.host, self.port)

        self.transport.close()

```

From what I've been able to tell, `Empty()` no longer exists, because the garbage collector already disposes it. Would adding such an if test to `format`, like the guys from happybase did,  be the way to go?

**Comment by xolox** [comment]
Hi George and thanks for the clear explanation. I haven't been able to reproduce the problem you reported and in fact I can't even find clear relevant details about how Python's garbage collection works. Nevertheless I did push a new release (7.0.1) that _intends_ to fix this, if the following assumption of mine is correct:

So `Empty` is `None` because module variables have already been garbage collected. What I assume but can't confirm from documentation is that this will either happen to all of the module variables or it will happen to none of them. In other words, all module variables are set to `None` at the same time and Python code won't get to see a situation where some but not all variables have been set to `None`.

This is important because I added an `if Empty is not None` check, but after that check there are references to other module variables as well and I don't feel like 'validating' all of those. So I was hoping that `if Empty is not None` is also an indication that the other module variables are save to use.

An alternative approach could have been to use a `try` / `except` block, but using one of those when I don't even know exactly which error conditions I want to recognize and ignore feels very wrong :slightly_smiling_face: .
